doc/source/building/working_with_images/vagrant_setup.rst

```diff
@@ -61,6 +53,46 @@ the following changes are required:
    provider including a pre-defined disk size. The disk size is
    optional, but recommended to provide some free space on disk.
 
+   For the VirtualBox provider, there is the additional attribute
+   ``virtualbox_guest_additions_present``, which you can set to ``true``
+   when the VirtualBox guest additions are installed in the KIWI
+   image:
+
+   .. code:: xml
+
+      <type image="vmx" filesystem="ext4" format="vagrant" boottimeout="0">
+          <vagrantconfig provider="virtualbox" virtualbox_guest_additions_present="true" virtualsize="42"/>
+          <size unit="G">42</size>
+      </type>
+
+   The resulting vagrant box will then use the ``vboxfs`` module for the
+   synchronized folder instead of rsync, that is used by default.
+
+2. Add mandatory packages
+
+   .. code:: xml
+
+      <package name="sudo"/>
+      <package name="openssh"/>
+
+   You also must include ``rsync`` for the libvirt provider and for the
+   VirtualBox provider with guest additions disabled:
+
+   .. code:: xml
+
+      <package name="rsync"/>
+
+   Please install the VirtualBox guest additions when these have been
+   enabled for the VirtualBox provider. KIWI cannot verify that these are
+   installed and thus not installing them will result in a broken box. For
+   example for openSUSE add these packages:
+
+   .. code:: xml
+
+      <package name="virtualbox-guest-tools"/>
+      <package name="virtualbox-guest-x11"/>
+      <package name="virtualbox-guest-kmp-default"/>
```

Comment:
Are these package mandatory? I'm not sure if this is "just an example" (the reader can ignore it) **or** it has to be done by the reader.

Reply:
This is related to the `virtualbox_guest_additions_present` setting. If you build a VirtualBox vagrant box and set `virtualbox_guest_additions_present` to true, then you have to include the virtualbox guest additions (on opensuse that are the above packages). If you build a libvirt box, or don't set `virtualbox_guest_additions_present` then you can simply ignore that.

That was probably not obvious from the text, do you have an idea how to rephrase/restructure it?

Reply:
Oh I see! In that case, I would split the step into two: first add the (real) mandatory packages. Second, if you set this attribute to true, add the packages a, b, and c into the KIWI XML file.

So basically, the structure could look like this:

```
2.  Add mandatory packages
   [ ... ]

3. If you have set the attribute `virtualbox_guest_additions_present` to true, add the following packages:
   [...]
```

Reply:
That's a good idea, thanks! I've restructured the text accordingly.